packages/contracts-ts/src/estimateGas.ts

```diff
@@ -0,0 +1,104 @@
+import { gasPriceOracleABI, gasPriceOracleAddress } from "./constants"
+import { getContract, createPublicClient, http, BlockTag } from 'viem'
+import * as chains from 'viem/chains'
+
+
+type Bytes = `0x${string}`
+
+// Generated addresses don't generate mainnet at this moment
+type BlockOptions = {
+  blockNumber?: bigint | undefined
+  blockTag?: BlockTag | undefined
+}
+
+type ProviderOptions = {
+  chainId: keyof typeof gasPriceOracleAddress
+  rpcUrl?: string | undefined
+  nativeCurrency?: chains.Chain['nativeCurrency']
+} | {
+  chainId: number
+  rpcUrl: string
+  nativeCurrency?: chains.Chain['nativeCurrency']
+}
+
+const warnAboutFetch = () => {
+  if (typeof fetch === 'undefined') {
+    console.error('No fetch implementation found. Please provide a fetch polyfill. This can be done in NODE by passing in NODE_OPTIONS=--experimental-fetch or by using the isomorphic-fetch npm package')
```

Comment:
```suggestion
    throw new Error('No fetch implementation found. Please provide a fetch polyfill. This can be done in NODE by passing in NODE_OPTIONS=--experimental-fetch or by using the isomorphic-fetch npm package')
```